File: benchmarks/nyc_taxi/r/q1_long_trips.R
args <- commandArgs(trailingOnly = TRUE)
if (length(args) < 1) {
  stop("usage: Rscript q1_long_trips.R <parquet-dataset-path>", call. = FALSE)
}

suppressPackageStartupMessages({
  library(arrow)
  library(dplyr)
})

started <- Sys.time()
dataset_path <- args[[1]]
ds_tbl <- open_dataset(dataset_path, format = "parquet")

result <- ds_tbl |>
  filter(trip_distance > 10) |>
  summarise(avg_fare = mean(fare_amount, na.rm = TRUE), trips = n()) |>
  collect()

print(result)
cat("ROWS_SCANNED=NA\n")
cat(sprintf("ROWS_RETURNED=%s\n", nrow(result)))
cat(sprintf("ELAPSED_SEC=%.6f\n", as.numeric(difftime(Sys.time(), started, units = "secs"))))

File: benchmarks/nyc_taxi/python/q1_long_trips.py
import argparse
import time

import pyarrow.compute as pc
import pyarrow.dataset as ds


def main() -> None:
    parser = argparse.ArgumentParser()
    parser.add_argument("--dataset", required=True, help="Path to the partitioned parquet dataset")
    args = parser.parse_args()

    started = time.perf_counter()
    dataset = ds.dataset(args.dataset, format="parquet", partitioning="hive")

    table = dataset.to_table(
        filter=pc.field("trip_distance") > 10,
        columns=["fare_amount"],
    )
    frame = table.to_pandas()
    avg_fare = frame["fare_amount"].mean()
    rows_returned = 1

    print(f"avg_fare={avg_fare}")
    print("ROWS_SCANNED=NA")
    print(f"ROWS_RETURNED={rows_returned}")
    print(f"ELAPSED_SEC={time.perf_counter() - started:.6f}")


if __name__ == "__main__":
    main()
